File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    
    total = brown + yellow
    
    # i : 가로
    for i in range(3, total):
        # j : 세로
        for j in range(3, i+1): # 세로는 가로와 같거나 작음
            if i * j == total:
                if (i-1)*2 + (j-1)*2 == brown:
                    return [i, j]
    

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        int total = brown + yellow;
        
        for (int h = 3; h < (int)(Math.sqrt(total)) + 1; h++) {
            if (total % h != 0) continue;
            
            int w = total / h;
            int b = w * 2 + (h - 2) * 2;
            if (b == brown) {
                answer[0] = w;
                answer[1] = h;
                break;
            }
        }
        return answer;
    }
}